File: src/BaseHmiTypes/Scripts/HmiScript.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Scripts;

public enum HmiScriptType
{
    None,
    Sub,
    Function
}

public sealed class HmiScript : IHmiObject
{
    public DateTime? LastModified { get; set; }

    public string? Name { get; set; }

    public HmiMultilingualText? Comment { get; set; }

    public HmiScriptLanguage Language { get; set; } = HmiScriptLanguage.Unknown;

    public string? SourceCode { get; set; }

    public string? PreCode { get; set; }

    public HmiScriptType ScriptType { get; set; } = HmiScriptType.None;
}

File: typescript/base-hmi-types/src/scripts/HmiScript.ts
import { IHmiObject } from "../IHmiObject.js";
import { HmiMultilingualText } from "../common/HmiMultilingualText.js";
import { HmiScriptLanguage } from "../screens/base/HmiProperty.js";

export enum HmiScriptType {
  None = "None",
  Sub = "Sub",
  Function = "Function",
}

export class HmiScript implements IHmiObject {
  lastModified?: Date;
  name?: string;
  comment?: HmiMultilingualText;
  language: HmiScriptLanguage = HmiScriptLanguage.Unknown;
  sourceCode?: string;
  preCode?: string;
  scriptType: HmiScriptType = HmiScriptType.None;
}
